up-l1/README.adoc

```diff
@@ -67,7 +67,7 @@ NOTE: uProtocol data model used by the uTransport interface are declared in http
 `UListener` is an interface used for receiving messages by the client uE from the uTransport layer. Listeners are used for receiving messages asynchronously by the uTransport layer.  
 
   * Type name *MUST* be `UListener`
-  * *MUST* contain the method `UStatus onReceive(UUri&, UPayload&, UAttributes&)` that is called by the Transport to notify (callback) the client
+  * *MUST* contain the method `UStatus onReceive(UMessage&)` that is called by the Transport to notify (callback) the client
```

Comment:
does this mean that the listener can return a `UStatus` to the transport implementation? Is there a particular expectation how that should be used and what the transport implementation should do about it?

Reply:
@sophokles73 this has been a long standing debate and there are two school of thoughts from developers, some want to return (so the operations can be part of a flow) and others say "it is a callback, it shouldn't do anything". I would love to hear others opinions but to this date there is no defined behavior as to what we should do if the return is anything but OK.

Reply:
Done, I checked and up-java returns void anyways